### PR TITLE
Jormungandr: add impacted objects to disruptions

### PR DESCRIPTION
**Pull request by antoine-de**
Add impacted_objects to the /disruptions API.

the goal is to have:

``` json
{
    "disruptions": 
    [
        {
            "status": "past",
            "application_periods": 
            [
                {
                    "begin": "20151201T060400",
                    "end": "20151201T082259"
                }
            ],
            "updated_at": "20151202T090248",
            "uri": "3eabf0a6-2bd6-472f-94c4-3aedd83caaa5",
            "contributor": "ire sncf",
            "severity": 
            {
                "color": "#000000",
                "priority": 42,
                "name": "trip delayed",
                "effect": "DETOUR"
            },
            "cause": "a problem",
            "impacted_objects": [
                {
                    "pt_object": {
                        "vehicle_journey": {
                            "uri": "bob"
                            ...
                        }
                    },
                    "impacted_stops": [
                        {
                            "stop_point": { "uri": "B", ... },
                            "base_departure": 171500,
                            "base_arrival": 170000,
                            "cause": "cochon sur les voies",
                            "stoptime_effect": "DELETED"
                        },
                        {
                            "stop_point": { "uri": "E", ... },
                            "departure": 171500,
                            "arrival": 170000,
                            "cause": "cochon sur les voies",
                            "stop_timeeffect": "ADDED"
                        },
                        {
                            "stop_point": { "uri": "C", ... },
                            "departure": 181700,
                            "arrival": 181700,
                            "base_departure": 180000,
                            "base_arrival": 180000,
                            "cause": "vache sur les voies",
                            "stop_timeeffect": "SIGNIFICANT_DELAYS"
                        },
                        {
                            "stop_point": { "uri": "D", ... },
                            "departure": 192300,
                            "arrival": 192300,
                            "base_departure": 190000,
                            "base_arrival": 190000,
                            "cause": "vache sur les voies",
                            "stop_timeeffect": "SIGNIFICANT_DELAYS"
                        }
                    ]
                }
            ]
        },
        {
            "status": "past",
            "application_periods": 
            [
                {
                    "begin": "20151201T060400",
                    "end": "20151201T082259"
                }
            ],
            "updated_at": "20151202T090248",
            "uri": "3eabf0a6-2bd6-472f-94c4-3aedd83caaa6",
            "contributor": "transilien",
            "severity": 
            {
                "color": "#000000",
                "priority": 42,
                "name": "trip delayed",
                "effect": "NO_SERVICE"
            },
            "cause": "another problem",
            "impacted_objects": [
                {
                    "pt_object": {
                        "line": {
                            "uri": "bobette",
                            ...
                        }
                    }
                },
                {
                    "pt_object": {
                        "stop_area": {
                            "uri": "bobitto",
                            ...
                        }
                    }
                }
            ],
            "messages": [
                {
                    "text": "<p>_pas_de_filtre<br></p>",
                    "channel": 
                    {
                        "content_type": "text/html",
                        "types": ["web"],
                        "name": "we"
                    }
                },
                {
                    "text": "_pas_de_filtre",
                    "channel": 
                    {
                        "content_type": "text/plain",
                        "types": ["email"],
                        "name": "email"
                    }
                }
            ]
        }
    ]
}
```

but for the moment the 'impacted_stops' are not outputed

### CODE REVIEW

**Comment by antoine-de** [comment]
we cannot merge it yet,because we need to merge https://github.com/CanalTP/navitia-proto/pull/25 before

File: source/jormungandr/tests/check_utils.py [hunk 1]
```diff
@@ -842,6 +842,10 @@ def is_valid_vehicle_journey(vj, depth_check=1):
         assert 'validity_pattern' not in vj
 
 
+def is_valid_trip(trip, depth_check=1):
```
Comment (xlqian) [inline source/jormungandr/tests/check_utils.py:845]: `depth_check` is not used...

Comment (antoine-de) [inline source/jormungandr/tests/check_utils.py:845]: for the moment, but it will change :wink: And those methods needs to have the same signature

Comment (xlqian) [inline source/jormungandr/tests/check_utils.py:845]: Ok! :grin:

**Comment by TeXitoi** [comment]
OK for me.

File: source/type/pb_converter.cpp [hunk 2]
```diff
@@ -121,15 +121,48 @@ static void fill_property(const std::string& name,
     property->set_value(value);
 }
 
+struct PtObjVisitor: public boost::static_visitor<> {
+    const nt::Data& data;
+    pbnavitia::PtObject* pb_pt_pbj;
```
Comment (pbougue) [inline source/type/pb_converter.cpp:126]: I guess you wanted to write `pb_pt_obj` (same after)

File: source/type/pb_converter.cpp [hunk 3]
```diff
@@ -465,7 +512,8 @@ void fill_pb_object(nt::Line const* l, const nt::Data& data,
 
 void fill_pb_object(const nt::LineGroup* lg, const nt::Data& data,
         pbnavitia::LineGroup* line_group, int max_depth,
-        const pt::ptime& now, const pt::time_period& action_period, const bool show_codes){
+        const pt::ptime& now, const pt::time_period& action_period
+                    , const bool show_codes, const DumpMessage dump_message) {
```
Comment (pbougue) [inline source/type/pb_converter.cpp:516]: indent

File: source/type/pb_converter.h [hunk 4]
```diff
@@ -208,7 +226,7 @@ void fill_pb_object(const georef::POIType*, const type::Data &data, pbnavitia::P
 void fill_pb_object(const georef::Admin* adm, const type::Data& data, pbnavitia::AdministrativeRegion* admin, int max_depth = 0,
```
Comment (pbougue) [inline source/type/pb_converter.h:226]: line length

File: source/type/pb_converter.h [hunk 5]
```diff
@@ -208,7 +226,7 @@ void fill_pb_object(const georef::POIType*, const type::Data &data, pbnavitia::P
...
                     const boost::posix_time::ptime& now = boost::posix_time::not_a_date_time,
                     const boost::posix_time::time_period& action_period = null_time_period,
-                    const bool show_codes=false);
+                    const bool show_codes = false, const DumpMessage = DumpMessage::Yes);
 
 void fill_pb_object(const navitia::type::StopTime* st, const type::Data& data, pbnavitia::ScheduleStopTime* row, int max_depth = 0,
```
Comment (pbougue) [inline source/type/pb_converter.h:231]: line length

**Pull request merged**